MaterialLib/PorousMedium/Porosity/createPorosityModel.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file:   createPorosityModel.h
+ *
+ * Created on August 16, 2016, 1:16 PM
+ */
+
+#ifndef OGS_CREATEPOROSITYMODEL_H
+#define OGS_CREATEPOROSITYMODEL_H
+
+#include <memory>
+
+#include "Porosity.h"
+
+#include "BaseLib/ConfigTree.h"
+
+namespace MaterialLib
+{
+namespace PorousMedium
+{
+/// Create a porosity model
+/// \param config  ConfigTree object has a tag of <porosity>
+std::unique_ptr<Porosity> createPorosityModel(BaseLib::ConfigTree const& config);
```

Comment:
Not sure here, but would it be possible just to forward declare Porosity, instead of including the header?
Definitely for the ConfigTree a forward declaration is sufficient.